pkg/collector/corechecks/system/wincrashdetect/probe/wincrashprobe.go

```diff
@@ -11,38 +11,128 @@ import (
 	"fmt"
 	"os"
 	"path/filepath"
+	"sync"
 
 	sysconfigtypes "github.com/DataDog/datadog-agent/cmd/system-probe/config/types"
 	"github.com/DataDog/datadog-agent/pkg/util/winutil"
 	"golang.org/x/sys/windows/registry"
 )
 
+type probeState uint32
+
+const (
+	// Idle indicates that the probe is waiting for a request
+	idle probeState = iota
+
+	// Busy indicates that the probe is currently processing a crash dump
+	busy
+
+	// Completed indicates that the probe finished processing a crash dump.
+	completed
+
+	// Failed indicates that the probe failed to process a crash dump.
+	failed
+)
+
 // WinCrashProbe has no stored state.
 type WinCrashProbe struct {
+	state  probeState
+	status *WinCrashStatus
+	mu     sync.Mutex
 }
 
 // NewWinCrashProbe returns an initialized WinCrashProbe
 func NewWinCrashProbe(_ *sysconfigtypes.Config) (*WinCrashProbe, error) {
-	return &WinCrashProbe{}, nil
+	return &WinCrashProbe{
+		state:  idle,
+		status: nil,
+	}, nil
+}
+
+// Handles crash dump parsing in a separate thread since this may take very long.
+func (p *WinCrashProbe) parseCrashDumpAsync() {
+	if p.status == nil {
+		p.state = failed
+		return
+	}
+
+	parseCrashDump(p.status)
+
+	p.mu.Lock()
+	defer p.mu.Unlock()
+	p.state = completed
 }
 
 // Get returns the current crash, if any
 func (p *WinCrashProbe) Get() *WinCrashStatus {
 	wcs := &WinCrashStatus{}
 
-	err := wcs.getCurrentCrashSettings()
-	if err != nil {
-		wcs.ErrString = err.Error()
-		wcs.Success = false
-		return wcs
-	}
+	// Nothing in this method should take long.
+	p.mu.Lock()
+	defer p.mu.Unlock()
+
+	switch p.state {
+	case idle:
+		if p.status == nil {
+			// This is a new request.
+			err := wcs.getCurrentCrashSettings()
+			if err != nil {
+				wcs.ErrString = err.Error()
+				wcs.StatusCode = WinCrashStatusCodeFailed
+			}
+		} else {
+			// Use cached settings.
+			wcs.FileName = p.status.FileName
```

Comment:
could this whole block be
`wcs = p.Status`?

Reply:
and, is the `else` clause necessary? Is it possible to have the state be `idle` but have a non-nil `p.status`?

Reply:
> could this whole block be `wcs = p.Status`?
Not with `wcs.getCurrentCrashSettings()`, the first if clause. Also, I didn't know that Go did accept the C-style `?` conditional assignment. 

> and, is the `else` clause necessary? Is it possible to have the state be `idle` but have a non-nil `p.status`?
Not normally in production code, but tests do make use of this (see `cp.SetCachedSettings(crashStatus)`).  My first attempt at the tests made use of multiple implementations of `getCurrentCrashSettings` (production and test), but this `else` clause made the changes simpler and less code to manage.